File: python/LeetCode/00075.py
from typing import List

def sortColors(self, nums: List[int]) -> None:
    """
    Do not return anything, modify nums in-place instead.
    """
    r, w, b = 0, 0, 0

    for colour in nums:
        if colour == 0:
            r += 1
        elif colour == 1:
            w += 1
        elif colour == 2:
            b += 1
        else:
            raise ValueError()

    nums[:] = [0]*r + [1]*w + [2]*b

    return nums

File: go/leetcode/00075.go
package main

func sortColors(nums []int) {
	n := len(nums)

	red := 0
	white := 0
	for _, num := range nums {
		if num == 0 {
			red += 1
		} else if num == 1 {
			white += 1
		}
	}

	for i := 0; i < red; i++ {
		nums[i] = 0
	}

	for i := red; i < red+white; i += 1 {
		nums[i] = 1
	}

	for i := red + white; i < n; i += 1 {
		nums[i] = 2
	}
}
